### PR TITLE
MLPAB-694: Content updates for enter your reference

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

Content and validation error updates for `/certificate-provider-enter-reference-number`.

Fixes MLPAB-694

### CODE REVIEW

File: web/template/certificate_provider_enter_reference.gohtml [hunk 1]
```diff
@@ -12,7 +12,7 @@
             <form novalidate method="post">
```
[the remark this answers is not in the document]
Comment (acsauk) [reply web/template/certificate_provider_enter_reference.gohtml:12]: I was going to raise this content today - we have no link to the donor/LPA at this point so can't grab it from LPA details.

File: web/template/certificate_provider_enter_reference.gohtml [hunk 2]
```diff
@@ -12,7 +12,7 @@
...
                 <div class="govuk-form-group">
                     <fieldset class="govuk-fieldset">
-                        {{ template "input" (input . "reference-code" "referenceCode" .Form.ReferenceCode "classes" "govuk-input--width-20 govuk-!-margin-bottom-3")}}
+                        {{ template "input" (input . "reference-code" "certificateProviderReference" .Form.ReferenceCode "classes" "govuk-input--width-20 govuk-!-margin-bottom-3" "context" "certificateProviderReferenceContext")}}
```
[the remark this answers is not in the document]
Comment (acsauk) [reply web/template/certificate_provider_enter_reference.gohtml:15]: Good spot - will update.

File: web/template/certificate_provider_enter_reference.gohtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hawx) [reply web/template/certificate_provider_enter_reference.gohtml:12]: lol, great point

File: app/internal/page/certificateprovider/enter_reference_number.go [hunk 3]
```diff
@@ -14,30 +14,31 @@ import (
 	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
 )
 
-type enterReferenceCodeData struct {
+type enterReferenceNumberData struct {
```
Comment (hawx) [inline app/internal/page/certificateprovider/enter_reference_number.go:17]: any reason you changed to reference number? ABCDEF123456 isn't really a number

Comment (acsauk) [reply app/internal/page/certificateprovider/enter_reference_number.go:17]: Just including here for future reference as I know we covered on slack:

> Content flagged they'd agreed to use it taking in to account things like NI number, Vehicle reg number and driving licence number all containing letters and gave a term that most people would understand (figma was updated yesterday to reflect this)

**Review by hawx — APPROVED** [review]

**Pull request merged**